### ISSUE TITLE
Node 12 support

### ISSUE THREAD

**Issue by mo4islona** [znerol/node-xmlshim#13]
Seems `libxmljs` finally got Node12

https://github.com/libxmljs/libxmljs/commit/5131a387147b839c2030c089c4ea11a7d6bd1cb0

But where is no release yet. Maybe we can fetch code from master branch?

**Comment by mo4islona** [comment]
```{    
dependencies": {
    "jsdom": "^8.5.0",
    "libxmljs": "https://github.com/libxmljs/libxmljs"
},
```

This builds correctly on Node12

```yarn install v1.17.0
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 65.93s.
```

But many tests are failed

```domimplTest
✔ testParseSimpleXML
✔ testParseSimpleXMLDefaultNS
✖ testSimpleRoundtrip

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.testSimpleRoundtrip (/Users/mo4islona/Projects/node-xmlshim/test/domimplTest.js:31:29)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/domimplTest.js:7:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✔ should have access to node type constants through node instances

domparserTest
✔ should parse single element non-namespace xml
✔ should parse elements with default namespace
✔ should parse elements with namespace prefix
✔ child elements should inherit namespaces from parent
✔ child elements may override namespaces from parent
✔ should parse non-namespace attributes
✔ should parse attributes with namespace uri
✔ should parse text node
✔ should parse comments
✔ should parse text node before empty element into the enclosing element

xmlserializerTest
✖ should write single element non-namespace xml

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write single element non-namespace xml (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:21:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✖ should write elements with namespace prefix

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write elements with namespace prefix (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:36:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✖ should write non-namespace attributes

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write non-namespace attributes (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:50:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✖ should write attributes with namespace uri

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write attributes with namespace uri (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:67:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✖ should write text node

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write text node (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:82:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✖ should write comments

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write comments (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:112:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✖ should write document fragments

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write document fragments (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:126:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✖ should write single element node

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write single element node (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:138:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21

✖ should write single text node

TypeError: xml.TextWriter is not a constructor
    at exports.XMLSerializer.serializeToString (/Users/mo4islona/Projects/node-xmlshim/index.js:24:22)
    at Object.exports.should write single text node (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:150:17)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:232:20
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:168:13
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:131:25
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:165:17
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:463:34
    at Object.exports.setUp (/Users/mo4islona/Projects/node-xmlshim/test/xmlserializerTest.js:11:5)
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/lib/core.js:260:35
    at /Users/mo4islona/Projects/node-xmlshim/node_modules/nodeunit/deps/async.js:458:21


FAILURES: 10/58 assertions failed (909ms)
```

**Comment by znerol** [comment]
Regrettably xmlshim still depends on a forked version of libxmljs. See PR libxmljs/libxmljs#493 for details.

**Comment by mo4islona** [comment]
Any chance to update it to Node 12?

**Comment by znerol** [comment]
Updated libxmljs fork and published xmlshim version 0.2.6. This is now compatible with node 12.
